presto-main/src/main/java/io/prestosql/execution/SqlTaskManager.java

```diff
@@ -145,16 +146,15 @@ public SqlTaskManager(
         SqlTaskExecutionFactory sqlTaskExecutionFactory = new SqlTaskExecutionFactory(taskNotificationExecutor, taskExecutor, planner, splitMonitor, config);
 
         this.localMemoryManager = requireNonNull(localMemoryManager, "localMemoryManager is null");
-        DataSize maxQueryUserMemoryPerNode = nodeMemoryConfig.getMaxQueryMemoryPerNode();
+        DataSize maxQueryMemoryPerNode = nodeMemoryConfig.getMaxQueryMemoryPerNode();
         DataSize maxQueryTotalMemoryPerNode = nodeMemoryConfig.getMaxQueryTotalMemoryPerNode();
         DataSize maxQuerySpillPerNode = nodeSpillConfig.getQueryMaxSpillPerNode();
 
-        DataSize maxQueryMemoryPerNode = nodeMemoryConfig.getMaxQueryMemoryPerNode();
         queryMaxMemoryPerNode = maxQueryMemoryPerNode.toBytes();
-        queryMaxTotalMemoryPerNode = maxQueryMemoryPerNode.toBytes();
+        queryMaxTotalMemoryPerNode = maxQueryTotalMemoryPerNode.toBytes();
```

Comment:
looks like a bug, good catch!

Reply:
@sopel39 not sure if there's an easy way to write test for this bugfix, since it'll require asserting failure for a manufactured query that reliably uses low user memory but high total memory - per node. any ideas? (the change looks straightforward though.)

Reply:
I don't think it's worthwhile to add that test, given that an existing memory pool test started failing after this refactor and that we'd have to either either add `@VisibleForTesting` getters to the memory limit fields or probe at their values (one fake query at a time) by triggering failures through their already `@VisibleForTesting MemoryTrackingContext`. At the end of the day, the behavior here is relatively simple and the bug was only introduced looked like a copy-pasta error, not a behavioral subtlety.

Just my opinion though, if the above test is needed to get this merged I'll add it.

Reply:
I think such test is important as these memory limits are vital to Presto. In fact, I wonder why nobody spotted it already

Reply:
Test added.